tools/build-agent-release.sh

```diff
@@ -0,0 +1,54 @@
+RELEASEDIR=releases
+TAG=$(date +%Y%m%d)-0.$(git log --pretty=format:'%h' -n 1)
+
+BUILDS=(
+  # FORMAT
+  # GOOS GOARCH BINSUFFIX
+  "darwin 386 darwin-i386-$TAG.prod"
+  "darwin amd64 darwin-amd64-$TAG.prod"
+  "linux 386 linux-i386-$TAG.prod"
+  "linux amd64 linux-amd64-$TAG.prod"
+  "windows 386 windows-i386-$TAG.prod.exe"
+  "windows amd64 windows-amd64-$TAG.prod.exe"
+)
+
+
+# update_releases invokes the update_release_json tool to record an agent update.
+# It expects to be called with the following arguments:
+# 1. The path to the releases.json file to update,
+# 2. The component to update,
+# 3. The tag for the new release and
+# 4. The path to the new agent binary.
+function update_releases() {
+  releasejson=$(go run tools/update_release_json.go \
+    -releases $RELEASEDIR/$1 \
+    -component $2 \
+    -tag $3 \
+    -binary $4 \
+    -sha256 $(cat $RELEASEDIR/$4 | openssl sha256))
+
+  echo "$releasejson" > $RELEASEDIR/$1
+}
+
+
+for build in "${BUILDS[@]}"; do
+  IFS=" " read -ra args <<< "$build"
+
+  echo "Building an agent for ${args[0]}/${args[1]}"
```

Comment:
I would encourage setting `targetarch=` here first, and then including it in both in this debug message and also the “failed” message below so that a simple build log grep for “failed” shows precisely what failed without having to dig for more context.